### DISCUSSION TITLE
Consider blocking login when too many devices are logged in

### DISCUSSION THREAD

**Discussion by turt2live** [element-hq/element-meta#567] [Ideas]
After about 30 devices it gets a bit ridiculous. Homeservers (and therefore the spec) should have an option to throw an error code back during login to say that the user needs to log out somewhere first.

**Comment by dbkr** [comment]
Even just doing this in riot might be enough, ie. if you have more than 10 devices, show an interstitial after login with the same UI we have in settings to list & delete them. Probably also a 'continue anyway' button.

Edit: ...which is to say that doing it at the HS or spec level might be overkill as you can't refuse logins if the user has too many devices already since if they can't access any of the existing logins to log devices out, they're screwed.

**Comment by turt2live** [comment]
hmm, true, though the problem this UI would try and solve is device proliferation which would still happen if the device actually gets created in the background.

If the user is unable to access their previous devices, they can request a password reset which logs them out?
